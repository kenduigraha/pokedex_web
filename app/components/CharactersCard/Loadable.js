/**
 *
 * Asynchronously loads the component for CharactersCard
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
