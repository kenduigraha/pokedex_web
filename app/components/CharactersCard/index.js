import React, { memo } from 'react';
import PropTypes from 'prop-types';
// import { Link } from 'react-router-dom';
import { Card } from 'antd';
import { POKEMON_IMAGE_URL } from 'utils/constants';
import CharactersCardCol from './styles';
const { Meta } = Card;

const getIdPokemon = url => url.split('/')[url.split('/').length - 2];

const CharactersCard = ({ data, span, loading, showModal }) => (
  <CharactersCardCol
    span={span}
    xxl={6}
    xl={span}
    lg={8}
    md={12}
    sm={24}
    xs={24}
    onClick={() => showModal(data.name)}
  >
    <Card
      loading={loading}
      hoverable
      cover={
        <img
          alt={data.name}
          src={`${POKEMON_IMAGE_URL}/${getIdPokemon(data.url)}.png`}
        />
      }
    >
      <Meta title={data.name} />
    </Card>
  </CharactersCardCol>
);

CharactersCard.defaultProps = {
  span: 6,
};

CharactersCard.propTypes = {
  data: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  showModal: PropTypes.func,
  span: PropTypes.number,
};

export default memo(CharactersCard);
