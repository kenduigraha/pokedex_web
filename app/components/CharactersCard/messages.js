/*
 * CharactersCard Messages
 *
 * This contains all the text for the CharactersCard component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.CharactersCard';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the CharactersCard component!',
  },
});
