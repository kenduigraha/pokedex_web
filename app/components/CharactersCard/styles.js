import styled from 'styled-components';
import { Col } from 'antd';

const CharactersCardCol = styled(Col)`
  margin-bottom: 30px;
`;

export default CharactersCardCol;
