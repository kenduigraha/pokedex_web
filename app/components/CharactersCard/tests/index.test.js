/**
 *
 * Tests for CharactersCard
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router-dom';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import CharactersCard from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<CharactersCard />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <BrowserRouter>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <CharactersCard data={{ name: '', url: '' }} loading={false} />
        </IntlProvider>
      </BrowserRouter>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <BrowserRouter>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <CharactersCard data={{ name: '', url: '' }} loading={false} />
        </IntlProvider>
      </BrowserRouter>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
