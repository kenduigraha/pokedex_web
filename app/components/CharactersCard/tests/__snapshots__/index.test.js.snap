// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharactersCard /> Should render and match the snapshot 1`] = `
<div
  class="ant-col ant-col-6 styles__CharactersCardCol-sc-1adlsw4-0 imzfEL ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-6"
>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable"
  >
    <div
      class="ant-card-cover"
    >
      <img
        alt=""
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
      />
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      />
    </div>
  </div>
</div>
`;
